include/micm/solver/backward_euler.hpp

```diff
@@ -29,38 +28,47 @@ namespace micm
 {
 
   /// @brief An implementation of the fully implicit backward euler method
-  template<class LinearSolverPolicy, class ProcessSetPolicy>
+  template<class LinearSolverPolicy, class RatesPolicy>
   class BackwardEuler
   {
     BackwardEulerSolverParameters parameters_;
     LinearSolverPolicy linear_solver_;
-    ProcessSetPolicy process_set_;
+    RatesPolicy rates_;
     std::vector<std::size_t> jacobian_diagonal_elements_;
-    std::vector<micm::Process> processes_;
 
    public:
+    /// @brief Solver parameters typename
+    using ParametersType = BackwardEulerSolverParameters;
+
     /// @brief Default constructor
+    /// @param parameters Solver parameters
+    /// @param linear_solver Linear solver
+    /// @param rates Rates calculator
+    /// @param jacobian Jacobian matrix
     BackwardEuler(
         BackwardEulerSolverParameters parameters,
```

Comment:
Can it be cons ref to avoid a copy of the arguement?
```suggestion
        const BackwardEulerSolverParameters& parameters,
```